services/src/main/java/org/exoplatform/news/service/NewsTargetingService.java

```diff
@@ -33,11 +34,31 @@
    */
   List<NewsTargetingEntity> getTargets();
 
+  /**
+   * Gets news targets by id
+   * @param newsId
+   * @return {@link List} of news targets by id
+   */
+  List<String> getTargetsByNewsId(String newsId);
+
   /**
    * Gets  referenced targets
    *
    * @return {@link List} of referenced targets
    */
   List<NewsTargetingEntity> getReferencedTargets();
 
+
+  /**
+   * Set news targets
+   *@param news
+   */
+  void  setNewsTargets(News news);
```

Comment:
In term of API definition, using `set` in method name is not good since it's used generally for accessors.
Besides, the input of the method should be the minimal information needed to make the operation instead of passing the whole `news` object. In other terms, you should by example change the parameters to the exact needed inputs such as `saveNewsTarget(String newsId, List<String> targets)`